### PR TITLE
docs(filter):initial FilterField ValueTemplate docs and Filter Refres…

### PR DESCRIPTION
**Pull request by NansiYancheva**

### CODE REVIEW

File: components/filter/overview.md [hunk 1]
```diff
@@ -60,6 +62,253 @@ The Blazor Filter provides parameters that allow you to configure the component:
 | `Class` | `string` | The class that will be rendered on the outermost element. |
 | `Value` | `CompositeFilterDescriptor` | Sets the value of the filter component. |
 
+## Reference and Methods
+
+The Filter methods are accessible through its reference.
```
Comment (dimodi) [inline components/filter/overview.md:67]:
```suggestion
The Filter exposes methods for programmatic operation. To use them, define a reference to the component instance with the `@ref` directive attribute.
```

File: components/filter/overview.md [hunk 2]
```diff
@@ -60,6 +62,253 @@ The Blazor Filter provides parameters that allow you to configure the component:
...
+
+| Method | Description | Use-case
+| --- | --- | --- |
+| `Refresh` | Redraws the component. | Update the component when programmatically changing it during runtime. |
```
Comment (dimodi) [inline components/filter/overview.md:71]:
```suggestion
| Method | Description |
| --- | --- |
| `Refresh` | Redraws the component. Use this method to update the component UI after changing its `Value` programmatically. |
```

File: components/filter/fields/templates.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
+---
+title: Templates
+page_title: FilterField - Templates
```
Comment (dimodi) [inline components/filter/fields/templates.md:3]:
```suggestion
page_title: Filter Field Templates
```

File: components/filter/fields/templates.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+description: Templates for the FilterField
```
Comment (dimodi) [inline components/filter/fields/templates.md:4]: The meta description should be 150 characters or more.

```suggestion
description: Learn how to configure and use Templates for the Filter component FilterField. The ValueTemplate allows users to input a filtering value in custom UI.
```

File: components/filter/fields/templates.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+slug: filter-filterfield-templates
+tags: telerik,blazor,filter,filterfields,templates
+published: True
+position: 10
+---
+
+# FilterField Templates
```
Comment (dimodi) [inline components/filter/fields/templates.md:11]: For consistency with https://docs.telerik.com/blazor-ui/components/filter/fields/operators

```suggestion
# Filter Field Templates
```

File: components/filter/fields/templates.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+You can customize the FilterFiled appearance and behavior through its templates.
```
Comment (dimodi) [inline components/filter/fields/templates.md:13]: Use backticks for inline code and API members.

```suggestion
You can customize the `FilterFiled` appearance and behavior through its templates.
```

File: components/filter/fields/templates.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+The FilterField can use templates for:
```
Comment (dimodi) [inline components/filter/fields/templates.md:15]:
```suggestion
The `FilterField` provides the following templates:
```

File: components/filter/fields/templates.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+* [Value Template](#value-template)
+
+
+
```
Comment (dimodi) [inline components/filter/fields/templates.md:20]:
```suggestion
```

File: components/filter/fields/templates.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+## Value Template
+
+The `ValueTemplate` allows you to customize the default value input field of a single Filter Field. You can replace the value input field with a different component.
```
Comment (dimodi) [inline components/filter/fields/templates.md:23]:
```suggestion
The `ValueTemplate` allows you to customize the default value editor of a single Filter Field. You can replace the default editor component or change the component settings.
```

File: components/filter/fields/templates.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+The `context` of the`ValueTemplate` is of type FilterFieldValueTemplateContext. You can get and set the `FilterDescriptor`, associated with the FilterField, through the `context` of the`ValueTemplate`.
+
+To use the Filter Field value template, add a `<ValueTemplate>` tag inside the [FilterField]({%slug filter-fields%}).
+
+````CSHMTL
```
Comment (dimodi) [inline components/filter/fields/templates.md:29]:
```suggestion
>caption Using FilterField ValueTemplate

````CSHMTL
```

File: components/filter/overview.md [hunk 11]
```diff
@@ -60,6 +62,253 @@ The Blazor Filter provides parameters that allow you to configure the component:
...
+
+        public string Name { get; set; } = string.Empty;
+
+        public int AgeInYears { get; set; }
+    }
+}
+````
+
+
 ## Events
```
Comment (dimodi) [inline components/filter/overview.md:312]: Move the Events section before Parameter.

**Review by dimodi — APPROVED** [review]

**Review by ikoevska — COMMENTED** [review]
Ignored. There are more recent updates to the articles.

**Pull request merged**